Main/Renamer_UpperLower.cpp

```diff
@@ -61,18 +62,40 @@ void Renamer_UpperLower::RenameList(BList *FileList) {
 		char	*tempStr = new char[laenge + 1];
 
 		uint8	Buchstabe, Byte0 = 0;
-		bool	MakeUpper = !bool(fUpperOrLower->Menu()->IndexOf(fUpperOrLower->Menu()->FindMarked()));
+		int32   choice = fUpperOrLower->Menu()->IndexOf(fUpperOrLower->Menu()->FindMarked());
+	
+		int32 isCap = 1;
+		
 		for (int j = 0; j < laenge; j++) {
 			Buchstabe = ListItem->fName.ByteAt(j);
 			if (Buchstabe>=0xc0) {
 				Byte0 = Buchstabe;
 			} else {
-				if (MakeUpper) {
-					if (((Buchstabe>=0x61) && (Buchstabe<=0x7a)) || ( (Byte0==0xc3) && (Buchstabe>=0xa0 && Buchstabe<=0xb6) || (Buchstabe>=0xb8 && Buchstabe<=0xbe)))
+				switch (choice){
+					case 0: if (((Buchstabe>=0x61) && (Buchstabe<=0x7a)) || ( (Byte0==0xc3) && (Buchstabe>=0xa0 && Buchstabe<=0xb6) || (Buchstabe>=0xb8 && Buchstabe<=0xbe)))
```

Comment:
can this be a function for example capitalizeLetter() that take Byte0 and Buchstabe as parameters and return the new Bunchstabe?

Reply:
@janus2 so do you want me to put the entire `switch` statement into a function? Or do you want me to use BeOS's functions?